lib/galaxy/model/__init__.py

```diff
@@ -6136,7 +6136,7 @@ def get_user(cls, pk):
 
     @classmethod
     def get_users_by_email(cls, email):
-        return cls.user_query().filter_by(email=email)
+        return cls.user_query().filter(func.lower(User.table.c.email) == email.lower())
```

Comment:
It would make me more comfortable if this raised an exception here if multiple results were returned.

Reply:
This happens here: https://github.com/python-social-auth/social-core/blob/6bbad65832f311108f222aa5d6d36ad0bd8e8bc5/social_core/pipeline/social_auth.py#L78.

Reply:
Ok, great.  It looks like the only other place this is used is https://github.com/galaxyproject/galaxy/blob/5e8264f9ff4b388bbd2ab990bd4074b638261d32/lib/galaxy/model/__init__.py#L6126, and it's also only to count > 0, so this is fine.  I was worried we were actually using the response from this in auth somewhere that N == 1 was expected.

Reply:
I definitely think it's a good idea to validate these concerns, could've caused a ton of issues down the line otherwise if people already have duplicate accounts.